include/class.filter.php

```diff
@@ -164,6 +165,59 @@ function useReplyToEmail() {
         return ($this->use_replyto_email);
     }
 
+    function disableFilters($object) {
+        switch (get_class($object)) {
+            case 'Topic':
+            $object_id = 'topic_id';
+            $FAClass = 'FA_AssignTopic';
+                break;
+            case 'Dept':
+                $object_id = 'dept_id';
+                $FAClass = 'FA_RouteDepartment';
+                break;
+            case 'Staff':
+                $object_id = 'staff_id';
+                $FAClass = 'FA_AssignAgent';
+                break;
+            case 'Team':
+                $object_id = 'team_id';
+                $FAClass = 'FA_AssignTeam';
+                break;
+            case 'SLA':
+                $object_id = 'sla_id';
+                $FAClass = 'FA_AssignSLA';
+                break;
+            case 'TicketStatus':
+                $object_id = 'status_id';
+                $FAClass = 'FA_SetStatus';
+                break;
+            case 'Email':
+                $object_id = 'from';
+                $FAClass = 'FA_SendEmail';
+                break;
+            case 'Canned':
+                $object_id = 'canned_id';
+                $FAClass = 'FA_AutoCannedResponse';
+        }
+        $id = $object->getId();
+        $actions = FilterAction::objects()
+            ->filter(array('type' => $FAClass::$type));
+        foreach($actions as $fa) {
```

Comment:
This will be expensive if there are lots actions of same type (think CM). Perhaps we can filter by configuration field - i.e `LIKE '%$object_id%'` - although that might be expensive as well. Just adding an observation here...

Reply:
Thinking on this one. I can do some research and play with it.

Reply:
Based on the time it takes the query to execute, it seems like on average, it goes slightly faster the way it is now, however, if we used ```LIKE '%$object_id%'```, we wouldn't have to use the $FAClass variable, so I think it'll just be a matter of preference here.